### PR TITLE
ltex-ls-plus: init at 18.2.0

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
This adds the `ltex-ls-plus` package, which is a language server for LanguageTool.
It is a fork of the `ltex-ls` package with additional features & bugfixes.

It seems like the original `ltex-ls` is no longer maintained. I've decided to create a new package instead of changing the previous package with updated link.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by FirelightFlagboy** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 357814`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ltex-ls-plus</li>
  </ul>
</details>

**Comment by pinpox** [comment]
The verson 18.3.0 is out, might be worth updating the PR.

File: pkgs/by-name/lt/ltex-ls-plus/package.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "ltex-ls-plus";
+  version = "18.2.0";
+
+  src = fetchurl {
+    url = "https://github.com/ltex-plus/ltex-ls-plus/releases/download/${version}/ltex-ls-plus-${version}.tar.gz";
+    sha256 = "sha256-8qxeaVGd2r3SLerCgHSpQWFbXhIngV2cdv/jWN7HHOE=";
+  };
```
Comment (pinpox) [inline pkgs/by-name/lt/ltex-ls-plus/package.nix:16]:
```suggestion
  version = "18.3.0";

  src = fetchurl {
    url = "https://github.com/ltex-plus/ltex-ls-plus/releases/download/${version}/ltex-ls-plus-${version}.tar.gz";
    sha256 = "sha256-TV8z8nYz2lFsL86yxpIWDh3hDEZn/7P0kax498oicls=";
  };
```

**Review by pinpox — CHANGES_REQUESTED** [review]
Update version to 18.3.0

**Comment by austinbutler** [comment]
FYI there's also these other open PRs for `ltex-lsp-plus`:

1. https://github.com/NixOS/nixpkgs/pull/342724
2. https://github.com/NixOS/nixpkgs/pull/363410

**Comment by pinpox** [comment]
@vinnymeller would you be interested in being added as maintainer to this package too? Since you maintained the original ltex-ls package it might make sense to add you here.


> FYI there's also these other open PRs for `ltex-lsp-plus`:
> 
>     1. [lltex-ls-plus: init at 17.0.1; vscode-extensions.ltex-plus.vscode-ltex-plus: init at 14.0.2 #342724](https://github.com/NixOS/nixpkgs/pull/342724)
> 
>     2. [ltex-ls-plus: init at 18.3.0 #363410](https://github.com/NixOS/nixpkgs/pull/363410)

I didn't see this PR and opened https://github.com/NixOS/nixpkgs/pull/363410. Closing as duplicate.

**Comment by pinpox** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 357814`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ltex-ls-plus</li>
  </ul>
</details>

**Review by pinpox — APPROVED** [review]

**Pull request merged**